src/tree/azure/CustomAzureResourceTreeDataProvider.ts

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureResource } from 'api/docs/vscode-azureresources-api';
+import * as vscode from 'vscode';
+import { AzExtResourceType, ResourceModelBase } from '../../../api/src/index';
+import { AzureResourceProviderManager } from '../../api/ResourceProviderManagers';
+import { azureExtensions } from '../../azureExtensions';
+import { showHiddenTypesSettingKey } from '../../constants';
+import { ext } from '../../extensionVariables';
+import { AzureSubscriptionProvider } from '../../services/SubscriptionProvider';
+import { settingUtils } from '../../utils/settingUtils';
+import { BranchDataItemCache } from '../BranchDataItemCache';
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+import { ResourceTreeDataProviderBase } from '../ResourceTreeDataProviderBase';
+import { TreeItemStateStore } from '../TreeItemState';
+import { AzureResourceGroupingManager } from './AzureResourceGroupingManager';
+import { GroupingItem } from './GroupingItem';
+
+const supportedResourceTypes: AzExtResourceType[] =
+    azureExtensions
+        .map(e => e.resourceTypes)
+        .reduce((a, b) => a.concat(...b), []);
+
+export class CustomAzureResourceTreeDataProvider extends ResourceTreeDataProviderBase {
+    private subscriptionProvider: AzureSubscriptionProvider | undefined;
+    private filtersSubscription: vscode.Disposable | undefined;
+    private statusSubscription: vscode.Disposable | undefined;
+
+    constructor(
```

Comment:
I feel like a decent amount of this code is copied from 'AzureResourceTreeDataProvider`. We should be able to make a base class and extend it.